### PR TITLE
Update release workflows

### PR DESCRIPTION
**Pull request by Link-**
- resolves https://github.com/github/c2c-actions-runtime/issues/2221
- resolves https://github.com/github/c2c-actions-runtime/issues/2194
- resolves https://github.com/github/c2c-actions-runtime/issues/2212

### Summary

Following the transfer of `actions-runner-controller` to the `actions` organisation, the controller and runner release workflows have been broken.

This PR intends to fix those workflows while introducing the least amount of changes. The fact that packages (all images pushed to GHCR) remain under [the old organisation](https://github.com/orgs/actions-runner-controller/packages) – which is intentional to avoid breaking end-user environments – introduced unnecessary complexity to these workflows.

As we work on improving our posture, this PR remedies the situation and provides a mechanism to publish changes until the permanent fixes are introduced.

The behaviour of these workflows is fully described here: https://github.com/actions-runner-controller/releases

### Caveats

For all of these workflows, the image pushing after build has not been tested (to avoid polluting registries). This needs to be verified after we merge.

### TODO before merge

- [x] Commit all TODO suggestions
- [x] Test & confirm the [publish-chart.yaml workflow](https://github.com/actions/actions-runner-controller/blob/c407b33040afd6ca6ad6347105a30f8a94ac8824/.github/workflows/publish-chart.yaml) is working as expected
- [x] Remove all references to `set-output`: [ref1](https://github.com/actions/actions-runner-controller/blob/c407b33040afd6ca6ad6347105a30f8a94ac8824/.github/workflows/publish-chart.yaml#L102)
- [x] Update `DOCKER_ACCESS_TOKEN` and `DOCKER_USER` secrets in this repo and in actions-runner-controller/releases

### CODE REVIEW

File: .github/workflows/publish-chart.yaml
Comment (Link-) [inline .github/workflows/publish-chart.yaml]:
```suggestion
      CHART_TARGET_REPO: actions-runner-controller.github.io
```

**TODO:** this is needs to be committed before merging this PR.

File: .github/workflows/publish-canary.yaml
Comment (Link-) [inline .github/workflows/publish-canary.yaml]:
```suggestion
  PUSH_TO_REGISTRIES: true
```
**TODO:** This needs to be committed before merging this PR.

File: .github/workflows/runners.yaml
Comment (Link-) [inline .github/workflows/runners.yaml]:
```suggestion
  PUSH_TO_REGISTRIES: true
```
**TODO:** This needs to be committed before merging this PR.

File: .github/workflows/publish-arc.yaml
Comment (Link-) [inline .github/workflows/publish-arc.yaml]: **TODO:** Replace `v1` with commit sha (version pinning)

File: .github/workflows/publish-chart.yaml
Comment (Link-) [inline .github/workflows/publish-chart.yaml]: **TODO:** Replace v1 with commit sha (version pinning)

File: .github/workflows/publish-canary.yaml
Comment (Link-) [inline .github/workflows/publish-canary.yaml]: **TODO:** Replace v1 with commit sha (version pinning)

File: .github/workflows/runners.yaml
Comment (Link-) [inline .github/workflows/runners.yaml]: **TODO:** Replace v1 with commit sha (version pinning)

File: .github/workflows/publish-chart.yaml
Comment (vanZeben) [inline .github/workflows/publish-chart.yaml]: Can we pin this version SHA as well please 🙏🏻

File: .github/workflows/publish-chart.yaml
Comment (Link-) [inline .github/workflows/publish-chart.yaml]:
```suggestion
          cp ${{ github.workspace }}/index.yaml ${{ env.CHART_TARGET_REPO }}/actions-runner-controller/index.yaml
```
The index.yaml file should be inside the actions-runner-controller folder

File: .github/workflows/runners.yaml
Comment (Link-) [inline .github/workflows/runners.yaml]:
```suggestion
  PUSH_TO_REGISTRIES: true
```

**TODO:** This needs to be committed before merging this PR.

**Review by mumoshu — DISMISSED** [review]
@Link- I'm continually impressed by your awesome work! LGTM.
Let me know if you need quick re-approval after you added the commits for the CHART_TARGET_REPO change before merging.

**Comment by Link-** [comment]
You're too kind @mumoshu I've committed the changes and would appreciate one final ✅

**Review by vanZeben — APPROVED** [review]

**Review by mumoshu — APPROVED** [review]
LGTM. Thanks again for your efforts ☺️

**Pull request merged**
